application/src/main/java/org/thingsboard/server/controller/plugin/TbWebSocketHandler.java

```diff
@@ -539,6 +539,11 @@ public void close(WebSocketSessionRef sessionRef, CloseStatus reason) throws IOE
         }
     }
 
+    @Override
+    public boolean isOpen(String sessionId) {
+        return externalSessionMap.containsKey(sessionId);
```

Comment:
I think we need to check `internalSessionMap`, since `wsSessionsMap` in the DefaultWebSocketService holds internal sessions.
I also think that we should additionally check the actual native WS session:
`sessionMd.session.isOpen()`.

@ashvayka Do you agree?

Reply:
I am not sure. If the session is removed from the map, it is no longer used.

Reply:
Just in case :)